autonomy/cli/scaffold_fsm.py

```diff
@@ -753,6 +753,172 @@ def _add_abstract_round_abci_if_not_present(ctx: Context) -> None:
         add_item(ctx, SKILL, abstract_round_abci_public_id)
 
 
+# Scaffolding of tests
+class RoundTestsFileGenerator(RoundFileGenerator):
+    """RoundTestsFileGenerator"""
+
+    FILENAME = "tests_" + ROUNDS_FILENAME
+
+    ROUNDS_FILE_HEADER = dedent(
+        """\
+        \"\"\"This package contains the tests for rounds of {FSMName}.\"\"\"
+
+        from typing import Any, Dict, List, Callable, Hashable
+        from dataclasses import dataclass, field
+
+        import pytest
+
+        # TODO: define and import specific payloads explicitly by name
+        from packages.{author}.skills.{skill_name}.payloads import *
+        from packages.{author}.skills.{skill_name}.rounds import (
+            Event,
+            SynchronizedData,
+            {non_degenerate_rounds}
+        )
+        from packages.valory.skills.abstract_round_abci.base import (
+            BaseTxPayload,
+        )
+        from tests.test_skills.test_abstract_round_abci.test_base_rounds import (
+            BaseRoundTestClass,
+            BaseOnlyKeeperSendsRoundTest,
+            BaseCollectDifferentUntilThresholdRoundTest,
+            BaseCollectSameUntilThresholdRoundTest,
+         )
+
+
+        @dataclass
+        class RoundTestCase:
+            \"\"\"RoundTestCase\"\"\"
+
+            initial_data: Dict[str, Hashable]
+            payloads: BaseTxPayload
+            final_data: Dict[str, Hashable]
+            event: Event
+            synchronized_data_attr_checks: List[Callable] = field(default_factory=list)
+
+
+        MAX_PARTICIPANTS: int = 4
+
+    """
+    )
+
+    BASE_CLASS = dedent(
+        """\
+        class Base{FSMName}RoundTestClass(BaseRoundTestClass):
```

Comment:
addressing https://github.com/valory-xyz/open-autonomy/pull/1279#discussion_r963086036

and respective base classes imported:
https://github.com/valory-xyz/open-autonomy/blob/303b0a6f841379434fc6468d6b306ff5629baa82/autonomy/cli/scaffold_fsm.py#L781-L786